src/app/views/sidebar/sample-queries/SampleQueries.tsx

```diff
@@ -48,7 +53,9 @@ export class SampleQueries extends Component<ISampleQueriesProps, any> {
   };
 
   public componentDidUpdate = (prevProps: ISampleQueriesProps) => {
+    const { sampleQueries } = this.state;
     if (prevProps.samples.queries !== this.props.samples.queries) {
+      this.groups = generateGroupsFromList(sampleQueries, 'category');
```

Comment:
Calculating it close to its point of need prevents having to track its location and state elsewhere in the page

Reply:
I removed it with 8c67a81bfe47a5011eb99bb9edb1028dd5591c7f since your suggestion had it closer together to where it was needed